src/js/structs/Node.js

```diff
@@ -22,6 +24,26 @@ class Node extends Item {
     return this.get("framework_ids");
   }
 
+  getStatus() {
+    const drainInfo = this.get("drain_info");
+    const deactivated = this.get("deactivated");
+    const active =
```

Comment:
In addition, active is a specific field. I wonder how it changes when drain_info is set.

Reply:
Might be worth a unit test in order to keep it clear

Reply:
i added a unit test reflecting the information i got from Joseph.